### PR TITLE
feat: updating actions to use repo associated resources

### PR DESCRIPTION
**Pull request by gausnes**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      - name: Lint
+        run: docker build --target lint .
+
+      - name: Build test image
+        run: docker build --target test -t test-container .
+
+      - name: Run tests in test container
+        run: |
+          docker run --rm test-container
```
Comment (gausnes) [inline .github/workflows/main.yml]: Probably should update this to running on the agent as well, lmk if you have an action to reference 😁

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        target: [ x86_64-pc-windows-gnu, x86_64-unknown-linux-musl ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Compile and release
+        uses: rust-build/rust-build.action@v1.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          RUSTTARGET: ${{ matrix.target }}
+          EXTRA_FILES: "README.md LICENSE"
```
Comment (gausnes) [inline .github/workflows/main.yml]: This seemed to do what I was looking for, looking to remove docker from the pipe image and run the exporter as a service: https://github.com/rust-build/rust-build.action

If folks would prefer to still use a docker container that should be a pretty straightforward update as well.

Comment (gausnes) [reply .github/workflows/main.yml]: Mmhmm, [not quite working](https://github.com/gausnes/prometheus_wireguard_exporter/runs/4003413075?check_suite_focus=true) with the initial configuration:

```
error: linker `x86_64-linux-musl-gcc` not found
  |
  = note: No such file or directory (os error 2)

error: could not compile `libc` due to previous error
```

**Review by BroderickCarlin — APPROVED** [review]

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          body: ""
+      - name: Upload Release Asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./target/${{ matrix.target }}/release/${{ github.event.repository.name }}
+          asset_name: ${{ github.event.repository.name }}-${{ matrix.target }}
+          asset_content_type: application/octet-stream
```
Comment (gausnes) [inline .github/workflows/main.yml:65]: Example Release:

https://github.com/gausnes/prometheus_wireguard_exporter/releases/tag/v0.0.2

Was able to download and start the agent on the pipe-dev node:

```shell
$ wget https://github.com/gausnes/prometheus_wireguard_exporter/releases/download/v0.0.2/prometheus_wireguard_exporter-x86_64-unknown-linux-gnu
$ chmod +x prometheus_wireguard_exporter-x86_64-unknown-linux-gnu
$ ./prometheus_wireguard_exporter-x86_64-unknown-linux-gnu
[2021-11-01T23:50:37Z INFO  prometheus_wireguard_exporter] prometheus_wireguard_exporter v3.5.1 starting...
[2021-11-01T23:50:37Z INFO  prometheus_wireguard_exporter] using options: Options { verbose: false, prepend_sudo: false, extract_names_config_files: None, interfaces: None, metric_attributes: MetricAttributeOptions { split_allowed_ips: false, export_remote_ip_and_port: false, handshake_timeout_seconds: None } }
[2021-11-01T23:50:37Z INFO  prometheus_wireguard_exporter] starting exporter on http://0.0.0.0:9586/metrics
[2021-11-01T23:50:37Z INFO  prometheus_exporter_base] Listening on http://0.0.0.0:9586/metrics
```

**Comment by gausnes** [comment]
bumping

**Review by stefanjbergh — APPROVED** [review]
lgtm

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+on:
+  push:
+    branches:
+      - main
+      - master
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - main
```
Comment (lbredeso) [inline .github/workflows/main.yml]: I only see a `master` branch on this repo, is `main` needed here?

**Review by elventear — APPROVED** [review]

**Pull request merged**